app/controllers/find/courses/accrediting_providers_controller.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Find
+  module Courses
+    class AccreditingProvidersController < Find::ApplicationController
+      def show
+        @course = provider.courses.includes(
+          :enrichments,
+          subjects: [:financial_incentive],
+          site_statuses: [:site]
+        ).find_by!(course_code: params[:course_code]&.upcase).decorate
+
+        render_not_found if @course.accrediting_provider.blank?
+        render_not_found unless @course.is_published?
+      end
```

Comment:
Is there a potential double render error here?

Reply:
Possibly, I'll change it. Good catch